src/LibraryUtilities.ts

```diff
@@ -566,48 +285,244 @@ export function updateSections(oldLayoutSpecs: any, newLayoutSpecs: any): void {
- * by splitting the name with '.', and put them under a specific section.
+ * This function will convert layoutElements to ItemData, and construct an array of IncludeItemPair,
+ * which stores an includeInfo and its parent item.
  * 
- * As an example, for two typeListNodes 'A.B.C' and 'A.B.D', the resulting nodes are:
- * - A
- *   |- B
+ * @param {LayoutElement[]} layoutElements 
+ * The layoutElements to be converted to ItemData
+ * 
+ * @param {IncludeItemPair[]} pairs
+ * The IncludeItemPairs to be contructed from
+ * 
+ * @param {ItemData} parentItem 
+ * The parentItem of layoutElements (if exist)
+ */
+function convertLayoutElementToItemData(
+    layoutElements: LayoutElement[],
+    pairs: IncludeItemPair[],
+    parentItem?: ItemData):
+    ItemData[] {
+
+    let results: ItemData[] = [];
+
+    for (let layoutElement of layoutElements) {
+        let layoutData = new ItemData(layoutElement.text);
+        layoutData.constructFromLayoutElement(layoutElement);
+
+        // Sections that don't show its header will be expanded by default.
+        if (layoutData.itemType === "section" && !layoutData.showHeader) {
+            layoutData.expanded = true;
+        }
+
+        if (parentItem) {
+            parentItem.appendChild(layoutData);
+        }
+
+        if (layoutElement.include.length > 0) {
+            for (let include of layoutElement.include) {
+                let pair = new IncludeItemPair();
+                pair.include = include;
+                pair.parentItem = layoutData;
+                pairs.push(pair);
+            }
+        }
+
+        if (layoutElement.childElements.length > 0) {
+            convertLayoutElementToItemData(layoutElement.childElements, pairs, layoutData);
+        }
+
+        results.push(layoutData);
+    }
+
+    return results;
+}
+
+/**
+ * This method will merge all typeListNodes under corresponding library items based on includeItemPairs.
+ * Each includeItemPair stores includeInfo and the parent item of this includeInfo.
+ * 
+ * When iterating through includeItemPairs, comparison will happen between fullyQualifiedName of typeListNode 
+ * and path of the include info of each pair.. 
+ * 
+ * If there is a match of them, which means they are exactly the same, this type node will be merged to 
+ * the parent item of the includeInfo. 
+ * 
+ * If there is not a match, but fullyQualifiedName includes path, this type node will
+ * first be built from name, and then merged to the parent item.
+ * For example, if fullayQualifiedName is A.B.C.D, path is A.B, the item to be merged would have the 
+ * following structure.
+ * 
+ * - parentItem
+ *  |- B
  *      |- C
- *      |- D
+ *          |- D
  * 
- * @param {TypeListNode[]} allNodes 
- * All the typeListNodes
+ * If there is not a match or include, skip to next includeInfo/typeListNode, depending on their 
+ * lexicographical order.
```

Comment:
This is really a good example of great documentation, keep up the good work, @fanwgwg 👍 I would as usual suggest some tweaks to the documentation as follow:

/**
 * This function merges all TypeListNode objects under the corresponding ItemData 
 * (i.e. a library item) based on information provided in IncludeItemPair. Each 
 * IncludeItemPair stores an IncludeInfo and the owning ItemData for that IncludeInfo.
 * 
 * When the function iterates through each IncludeItemPair object, comparison happens 
 * between fullyQualifiedName of a given TypeListNode, and the IncludeInfo.path found 
 * in each IncludeItemPair.
 * 
 * If fullyQualifiedName matches IncludeInfo.path, it means they are exactly the same 
 * and therefore the TypeListNode will be merged to the owning ItemData specified in 
 * the IncludeItemPair.
 * 
 * If IncludeInfo.path does not completely match fullyQualifiedName but represents a 
 * substring of fullyQualifiedName, then the following takes place:
 * 
 * Hierarchical structure of ItemData objects will be generated by looking at the 
 * fullyQualifiedName of TypeListNode. These ItemData objects will then be merged 
 * under the owning ItemData specified in the IncludeItemPair. For example: if the 
 * fullyQualifiedName is "A.B.C.D" and IncludeInfo.path is "A.B", then the generated 
 * ItemData take the following form:
 * 
 * - owningItem
 *    |- B
 *       |- C
 *          |- D
 * 
 * If there is no match between fullyQualifiedName and IncludeInfo.path, then skip to 
 * the next IncludeInfo/TypeListNode depending on their lexicographical order.
*/